### ISSUE TITLE
Deprecate and remove unnecessary emit method on EventEmitter

### ISSUE THREAD

**Issue by kolecko** [angular/angular#47205]
### Which @angular/* package(s) are relevant/related to the feature request?

core

### Description

It's been a while 'next' on EventEmitter was deprecated. KISS comes handy right now, adapter is doing nothing just calling super.next(), if it is required to do some additional operations in the future at next, it might be just override. EventEmitter can be still enforced to be used in certain cases.

Does somebody have any other reason than just extension of behaviour why it's done this way?

### Proposed solution

```
/**
   @deprecated
   * Emits an event containing a given value. Alias for 'next'.
   * @param value The value to emit.
 */
emit(value?: T): void;
```

### Alternatives considered

Update docs to explicitly say that this is or is not meant to be just alias for RxJS subject next.

**Comment by Xample** [comment]
Hi Kolecko, you wrote both:
> It's been a while 'next' on EventEmitter was deprecated
```
And:
@deprecated
emit(value?: T): void;
```

Which one do you want to keep ? `next` or `emit` ?

This part of Angular is built on top of rxjs, it was not initially intended to keep rxjs all the way long, which is the reason for this abstraction i.e. an `EventEmitter` not to be a Subject. But today, even in v14 we are still dealing with a hidden disguised `Subject` in `EventEmitter`.

Currently, [there is no plain / native](https://github.com/tc39/proposal-observable) JS asynchronous way to achieve the same as an observable. We might have thought about native [async generators](https://www.google.com/search?q=convert+an+observable+to+async+generator) (a generator emitting promises) but this is the consumer which asks for data and not the source which notifies the subscribers to handle the event. Other frameworks or libraries are having the same problem. We actually went through the same problem with `promises` 10 years ago, when the promises were not really native but still better than callbacks (remember [bluebird](https://www.npmjs.com/package/bluebird), [q](https://www.npmjs.com/package/q), and [all the others](https://www.npmjs.com/search?q=promises)).

In angular, you never subscribe yourself to an `EventEmitter`, angular does it for you. You cannot neither (unless you cheat) expose an arbitrary observable out of your component, you need to go through that
```typescript
@Output()
event: EventEmitter
```
 
If the angular team decided to cross the bridge and agree on their rxjs dependency, then we might have any output based on RxJS. For instance, this would be legal:

```typescript
@Output()
seconds$ =  interval(1000);
```

But generally speaking, this is not the goal of an HTML element to emit events without any user's interaction. Enforcing people to use an `EventEmitter` is therefore a way of doing so.

I did not dig back into the docs but as far as I can remember, I read that emit is an alias of next.

**Comment by kolecko** [comment]
I meant "emit" as I wrote. Next was deprecated there might had been some reason for it so it might confuse everybody that you change deprecated next immediately to deprecated emit...

**Comment by Xample** [comment]
Well, if you replace `emit` with `next` (i.e. you remove the EventEmitter with emit), you basically remove the abstraction layer, meaning you are then stuck with supporting rxjs "forever". This is the reason why we should not even consider the [`EventEmitter` to be a `Subject` at all](https://www.bennadel.com/blog/3038-eventemitter-is-an-rxjs-observable-stream-in-angular-2-beta-6.htm#comments_47949)

This abstraction layer is to prevent having breaking changes if the ng team wants to replace rxjs with anything else at some point. Rxjs is [backed by Microsoft](https://en.wikipedia.org/wiki/ReactiveX) angular [by Google](https://en.wikipedia.org/wiki/Angular_(web_framework)) (just to say they are different teams, because TS is also MS and first citizen of Angular…)

Clean coding is also preventing people doing things which could possibly break.

Related: https://github.com/angular/angular/issues/34394

**Comment by kolecko** [comment]
Some comment suggesting something from 2016 and still nothing simmilar is part of docs 
_(please read everything even alternate solutions I wrote in this issue)_
right now its not clean and uselessly complicated, its not preventing anything.

Subject is explicitly part of EventEmitter's public interface, so it´s not clear which one to use even with documentation.

**IMHO** emit is only kept because as you write abstraction, especially during compile time. So you cant assign X->Y and Y->X because ts compiler will tell you, that you are missing something.

**Comment by Xample** [comment]
ok ok, but again I would not recommend using `.next()` as the `EventEmitter` has the "undocumented" `next` public function only by inheritance.

On the document page: https://angular.io/api/core/EventEmitter

The only `next` appearing on the page is the parameter within the subscription because… yes `EventEmitter` is an observable and can be subscribed. This said, the [subscribe](https://angular.io/api/core/EventEmitter#subscribe) of `EventEmitter`, AFAIK is only here for internal angular purpose i.e. for the template factory to subscribe to the component's events and to call the targeted event's using the callback function i.e. `(myEvent)=myCallbackFunction()`

So to summarize, I would not recommend using `next` (and therefore not depreciate `emit`) but I would agree with explaining in the docs not to use the super method `next` it and to only rely on `emit`

**Comment by kolecko** [comment]
I would go for interface simplification... better than docs.

**Comment by pkozlowski-opensource** [comment]
> Well, if you replace emit with next (i.e. you remove the EventEmitter with emit), you basically remove the abstraction layer, meaning you are then stuck with supporting rxjs "forever". This is the reason why we should not even consider the [EventEmitter to be a Subject at all](https://www.bennadel.com/blog/3038-eventemitter-is-an-rxjs-observable-stream-in-angular-2-beta-6.htm#comments_47949)

@Xample is very well summarising our thinking on the topic. We want to keep the `EventEmitter` abstraction so people are not tied to RxJS implementation.
